tests/mock_crust/utils.rs

```diff
@@ -40,18 +40,15 @@ const BALANCED_POLLING: bool = true;
 pub fn gen_range_except<T: Rng>(rng: &mut T,
                                 low: usize,
                                 high: usize,
-                                exclude: Option<usize>)
+                                exclude: BTreeSet<usize>)
```

Comment:
This could be `exclude: &BTreeSet<usize>` instead, then we wouldn't need to clone the tree.